secant/Features/App/App.swift

```diff
@@ -43,6 +65,83 @@ extension AppReducer {
     )
     .debug()
 
+    private static let appReducer = AppReducer { state, action, environment in
+        switch action {
+            /// Checking presense of stored wallet in the keychain and presense of database files in documents directory.
+        case .checkWalletInitialisation:
+            var walletInitialisationState: InitialisationState = .uninitialized
+
+            let fileManager = FileManager()
+            // TODO: use updated dependency from PR #217 (https://github.com/zcash/secant-ios-wallet/pull/217)
+            let keysPresent = environment.recoveryPhraseStorage.areKeysPresent()
+            
+            do {
+                // TODO: Create a dependency to handle database files for the SDK, issue #220 (https://github.com/zcash/secant-ios-wallet/issues/220)
+                // TODO: use database URL from the same issue #220
+                let documentsURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+                let dataDatabaseURL = documentsURL.appendingPathComponent("ZcashSDK.defaultDataDbName", isDirectory: false)
+                let attributes = try fileManager.attributesOfItem(atPath: dataDatabaseURL.path)
+                let databaseFilesPresent = attributes.isEmpty
+                
+                switch (keysPresent, databaseFilesPresent) {
+                case (false, false):
+                    walletInitialisationState = .uninitialized
+                case (false, true):
+                    walletInitialisationState = .keysMissingFilesPresent
+                case (true, false):
+                    walletInitialisationState = .keysPresentFilesMissing
+                case (true, true):
+                    walletInitialisationState = .initialized
+                }
+            } catch CocoaError.fileNoSuchFile, CocoaError.fileReadNoSuchFile {
+                walletInitialisationState = keysPresent ? .keysPresentFilesMissing : .uninitialized
+            } catch {
+                walletInitialisationState = .failure(ErrorWrapper(error: error))
+            }
+            
+            return Effect(value: .walletInitialisationResult(walletInitialisationState))
+
+            /// All possible responses to the `.checkWalletInitialisation` action result.
+            /// `uninitialized` results in the start of the Onboarding flow for the user.
+        case .walletInitialisationResult(let walletInitialisationState):
```

Comment:
Do you think this could just be combined with the other `check` action and have the one action just return these effects or set this state?  If so the action could probably then be called something like `initialize`?

Reply:
I had all in one action and then I decided to split it. The first part of the action was deciding X (many possible state) and other part of the action was deciding Y based on X. I think by splitting it we can test both separately. Mock some states of the keychain and test checkWallet action. And separately test the 2nd action. Thoughts @adamstener ?

Reply:
It doesn't seem like it's benefiting anything to separate these out and its adding a bit of indirection.  It might be easier to understand what's going on here as an external reader if it was just inline.

Reply:
In the end you are right.. merged together, extra action removed